lib/jxl/decode_test.cc

```diff
@@ -1540,9 +1543,19 @@ TEST_P(DecodeTestParam, PixelTest) {
         xsize * 2 * orig_channels, nullptr, pixels.data(), pixels.size(),
         nullptr, nullptr, static_cast<jxl::Orientation>(config.orientation)));
   }
-
-  EXPECT_EQ(0u, ComparePixels(pixels.data(), pixels2.data(), xsize, ysize,
-                              format_orig, format));
+  if (config.upsampling == 1) {
+    EXPECT_EQ(0u, ComparePixels(pixels.data(), pixels2.data(), xsize, ysize,
+                                format_orig, format));
+  } else {
+    // resampling is of course not lossless, so as a rough check:
+    // count pixels that are more than off-by-25 in the 8-bit value of one of
```

Comment:
It says 25 in the comment but 50 in the code

Reply:
I think that factor is twice the error: if it is 2.0, then the error threshold is 1/255, so an off by one in uint8 terms.